File: js_ts/Leetcode/26_Remove_Duplicates_from_Sorted_Array.ts
function removeDuplicates(nums: number[]): number {
  let k = 0;
  for (let num of nums) {
    if (k < 1 || nums[k - 1] !== num) {
      nums[k] = num;
      k++;
    }
  }
  return k;
}

File: python/Leetcode/26_Remove_Duplicates_from_Sorted_Array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[k]:
                k += 1
                if k != i:
                    nums[k] = nums[i]                      
        return k + 1
    

print(Solution().removeDuplicates([1,1,2]))